### PR TITLE
added new cicd for sandbox lite

### PR DESCRIPTION
**Pull request by evalsocket**
Added CI for new sandbox lite

NOTE: Create release pipeline testing is not done

### CODE REVIEW

File: docker/sandbox-ultra/bootstrap/flyte/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
+apiVersion: apps/v1
```
Comment (wild-endeavor) [inline docker/sandbox-ultra/bootstrap/flyte/deployment.yaml:1]: let's remove this entire folder.  the whole flyte folder.

**Review by wild-endeavor — COMMENTED** [review]
left some comments, do they make sense?  i just don't like the idea of having two separate sources of truth.

**Comment by jeevb** [comment]
The entire k8s/kustomize dir can be dropped. I think the idea is to just copy the rendered helm chart in. That's all we really need.

File: docker/sandbox-ultra/Makefile [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	mkdir -p $$(dir $$@)/amd64
+	docker pull --platform linux/amd64 $(1)
+	docker tag $(1) $(word 1,$(subst :, ,$(1))):latest
+	docker save --output $$(dir $$@)amd64/$(subst /,-,$(word 1,$(subst :, ,$(1)))).tar $(word 1,$(subst :, ,$(1))):latest
+endef
+
+$(foreach name,$(IMAGES),$(eval $(call IMAGE_RULE,$(name))))
+
+images/tar/flyte:
```
Comment (jeevb) [inline docker/sandbox-ultra/Makefile:32]: We should be building/pushing this separately.

**Comment by evalsocket** [comment]
@jeevb In PR checks we are not publishing these images, In a release, we will re-use the single binary image.  

Other wise I need to use cache to store the image

File: docker/sandbox-ultra/Makefile (hunk 2 above)
Comment (evalsocket) [reply docker/sandbox-ultra/Makefile:33]: meaning ?

**Comment by wild-endeavor** [comment]
> The entire k8s/kustomize dir can be dropped. I think the idea is to just copy the rendered helm chart (single yaml file) in. That's all we really need.
> 
> Also looks like we are still building the Flyte container here. Can we just use a prebuilt one?

yeah let's build the flyte container as a separate job and then pull in here.

**Comment by evalsocket** [comment]
Release pipeline Test: 
Branch DIff: https://github.com/flyteorg/flyte/compare/cicd-sandbox-lite...evalsocket:flyte:master
Workflow: https://github.com/evalsocket/flyte/actions/runs/3320922597/jobs/5487909625

File: docker/sandbox-ultra/example/example.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import typing
```
Comment (wild-endeavor) [inline docker/sandbox-ultra/example/example.py:1]: we can get rid of this file.

File: docker/sandbox-ultra/Makefile [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+images/tar/flyte-ci:
+	mkdir -p $(dir $@)amd64
+	mkdir -p $(dir $@)arm64
+	cp /tmp/flyte-amd64.tar $(dir $@)amd64/flyte.tar
+	cp /tmp/flyte-arm64.tar $(dir $@)arm64/flyte.tar
+
+.PHONY: images
+images: $(foreach name,flyte $(IMAGES),images/tar/$(subst /,-,$(word 1,$(subst :, ,$(name)))))
+
+.PHONY: osxbuildflyte
```
Comment (wild-endeavor) [inline docker/sandbox-ultra/Makefile:48]: delete this

File: docker/sandbox-ultra/Makefile [hunk 5]
```diff
@@ -0,0 +1,106 @@
+.SILENT: help
+.PHONY: help
+help:
+	echo Available recipes:
+	cat $(MAKEFILE_LIST) | grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' | awk 'BEGIN { FS = ":.*?## " } { cnt++; a[cnt] = $$1; b[cnt] = $$2; if (length($$1) > max) max = length($$1) } END { for (i = 1; i <= cnt; i++) printf "  $(shell tput setaf 6)%-*s$(shell tput setaf 0) %s\n", max, a[i], b[i] }'
+	tput sgr0
+
+export DOCKER_BUILDKIT := 1
+
+IMAGES := \
```
Comment (jeevb) [inline docker/sandbox-ultra/Makefile:10]: We should probably just move this into a script.

File: docker/sandbox-ultra/Makefile [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+.PHONY: bootstrap
+bootstrap:
+	kubectl kustomize bootstrap > bootstrap/output.yaml
+
+.PHONY: clean
+clean: ## Cleanup state
+	rm -rf data
+
+.PHONY: build
+build: bootstrap images
```
Comment (jeevb) [inline docker/sandbox-ultra/Makefile:65]:
```suggestion
build: images
```

File: docker/sandbox-ultra/Makefile [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+osxbuildflyte:
+	echo hello
+
+.PHONY: images-ci
+images-ci: $(foreach name,flyte-ci $(IMAGES),images/tar/$(subst /,-,$(word 1,$(subst :, ,$(name)))))
+
+
+.PHONY: bootstrap
+bootstrap:
+	kubectl kustomize bootstrap > bootstrap/output.yaml
```
Comment (jeevb) [inline docker/sandbox-ultra/Makefile:58]:
```suggestion
```

**Comment by jeevb** [comment]
Nice work @evalsocket! This is very close. Let's just clean up the image pulling by moving it into a script?

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**